### PR TITLE
Provide webid

### PR DESCRIPTION
**Pull request by nicola**

### CODE REVIEW

**Comment by sible777** [comment]
Not for merge, just so that @nicola can take over :+1:

File: tls/index.js [hunk 1]
```diff
@@ -93,3 +97,132 @@ function verifyKey (certificate, uri, profile, mimeType, callback) {
...
+    cert.setExtensions([{
+        name: 'subjectAltName',
+        altNames: [{
+            type: 6, // URI
+            value: options.agent
+        }]
+    }])
+
+    // Should we self-sign the cert? For now I suppose.
+    cert.sign(cert.publicKey)
```
Comment (nicola) [inline tls/index.js]: this should be private key of a newly generated key

**Comment by sible777** [comment]
@nicola There seems to be an issue with the processing of the `spkac.cnf` file I generated for testing. When using node's `Certificate.exportPrivateKey` on the data it produces empty strings which then make it all the way down into forge's sign method and cause an error to throw.

I'm looking into this today, there isn't much information out there regarding handling spkac with node. It's also possible that the spkac file isn't valid.

**Comment by sible777** [comment]
generate still needs to pass some tests, particularly verification of the certificate.

**Comment by sible777** [comment]
@nicola The code passes on node version 0.12 but fails on 0.10. I will examine the logs later tonight if I get a chance, or you can.

Let me know if you notice anything wrong with the code. I think we still need to return some serialization of the cert other than a forge cert, so that it can be used in a browser.

**Comment by nicola** [comment]
please, go ahead from here: https://github.com/linkeddata/node-webid/pull/9 (create a new PR if needed)

**Pull request closed without merging**